src/net/ossl.cc

```diff
@@ -987,8 +987,20 @@ class session : public enable_shared_from_this<session>, public session_impl {
     session(session_type t, shared_ptr<tls::certificate_credentials> creds,
             std::unique_ptr<net::connected_socket_impl> sock, tls_options options = {})
       : _sock(std::move(sock))
-      , _local_address(fmt::to_string(_sock->local_address()))
-      , _remote_address(fmt::to_string(_sock->remote_address()))
+      , _local_address([this]() -> sstring {
+            try { 
+                return fmt::to_string(_sock->local_address());
+            } catch(...) {
+                return "0.0.0.0:00000";
+            }
+        }())
+      , _remote_address([this]() -> sstring {
+            try {
+                return fmt::to_string(_sock->remote_address());
+            } catch(...) {
+                return "0.0.0.0:00000";
```

Comment:
The catch should be tightened to `std::system_error`.

I'm not convinced by the fallback, the formatting doesn't account for family and protocol, but it's not obvious to me what else can be done here.